sdk/core/Azure.Core.TestFramework/src/TokenCredentialStub.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core.TestFramework
+{
+    public class TokenCredentialStub : TokenCredential
```

Comment:
This class was introduced because when I was investigating `BearerTokenAuthenticationPolicyTests` I noticed this logic was duplicated.

Reply:
I don't think we want to make this public/include it in the Test Framework. We can instead refactor the core tests to avoid duplicating. Feel free to submit an issue for this as it is now somewhat outside the scope of this PR (or feel free to fix it 😄 )

Reply:
I will revert this change.

Reply:
This is done now. I am going to remove the azure core label as well